ARCH/vpr_arch/k4_N8_tileable_reset_softadder_register_scan_chain_nonLR_caravel_io_skywater130nm.xml

```diff
@@ -0,0 +1,591 @@
+                <output name="lut4_out" num_pins="1"/>
+              </pb_type>
+              <pb_type name="carry_follower" blif_model=".subckt carry_follower" num_pb="1">
+                <input name="a" num_pins="1"/>
+                <input name="b" num_pins="1"/>
+                <input name="cin" num_pins="1"/>
+                <output name="cout" num_pins="1"/>
+              </pb_type>
+              <interconnect>
+                <direct name="direct1" input="frac_logic.in[0:1]" output="frac_lut4.in[0:1]"/>
+                <direct name="direct2" input="frac_logic.in[3:3]" output="frac_lut4.in[3:3]"/>
+                <direct name="direct3" input="frac_logic.cin" output="carry_follower.b"/>
+                <direct name="direct4" input="frac_lut4.lut2_out[1:1]" output="carry_follower.a"/>
+                <direct name="direct5" input="frac_lut4.lut2_out[0:0]" output="carry_follower.cin"/>
+                <direct name="direct6" input="carry_follower.cout" output="frac_logic.cout"/>
+                <!-- Xifan Tang: I use out[0] because the output of lut6 in lut6 mode is wired to the out[0] -->
+                <direct name="direct7" input="frac_lut4.lut4_out" output="frac_logic.out"/>
+                <mux name="mux2" input="frac_logic.cin frac_logic.in[2:2]" output="frac_lut4.in[2:2]"/>
+              </interconnect>
+            </pb_type>
+            <!-- Define flip-flop with scan-chain capability, DI is the scan-chain data input -->
+            <pb_type name="ff" blif_model=".subckt scff" num_pb="1">
+              <input name="D" num_pins="1"/>
+              <input name="DI" num_pins="1"/>
+              <input name="reset" num_pins="1"/>
+              <output name="Q" num_pins="1"/>
+              <clock name="clk" num_pins="1"/>
+              <T_setup value="66e-12" port="ff.D" clock="clk"/>
+              <T_setup value="66e-12" port="ff.DI" clock="clk"/>
+              <T_setup value="66e-12" port="ff.reset" clock="clk"/>
+              <T_clock_to_Q max="124e-12" port="ff.Q" clock="clk"/>
+            </pb_type>         
+            <interconnect>
+              <direct name="direct1" input="fabric.in" output="frac_logic.in"/>
+              <direct name="direct2" input="fabric.sc_in" output="ff.DI"/>
+              <direct name="direct4" input="ff.Q" output="fabric.sc_out"/>
+              <direct name="direct5" input="ff.Q" output="fabric.reg_out"/>
+              <complete name="complete1" input="fabric.clk" output="ff.clk"/>
+              <complete name="complete2" input="fabric.reset" output="ff.reset"/>
+              <mux name="mux1" input="frac_logic.out fabric.reg_in" output="ff.D">
+                <delay_constant max="25e-12" in_port="frac_logic.out" out_port="ff.D"/>
+                <delay_constant max="45e-12" in_port="fabric.reg_in" out_port="ff.D"/>
+              </mux>
+              <mux name="mux2" input="ff.Q frac_logic.out" output="fabric.out">
```

Comment:
Should we patch the pb_type so that the last LUT can output `cout` to regular outputs?

Reply:
@tangxifan, since we have only one output per LE in this architecture, should be instead have cout connected to regular routing by setting fc. In the fracturable LUt4 we could have used the second output to MUX cout there

Reply:
@tpagarani No problem. You just need to remove the `fc_override` lines for cin/cout in the architecture file.

Reply:
@tangxifan I am getting assertion with that in OpenFPGA compilation. Looks like cout can't be in both directlist as well as hooked up to regular routing. 

  Error 1: Fail to find port 'grid_clb[1][2].bottom_width_0_height_0__pin_38_'
/home/tpagarani/git/OpenFPGA/openfpga/src/fabric/build_top_module_directs.cpp:104 add_module_nets_tile_direct_connection: Assertion 'true == module_manager.valid_module_port_id(src_grid_module, src_port_id)' failed.
Aborted (core dumped)

Reply:
You are right. If a pin is accessible by regular routing architecture, it should be defined in the direct list.

Reply:
Hmm. However we need both in that case. Does it mean that we need to create an additional output pin for cout to connect to regular routing as well?

Reply:
Yes. You can refer to an [example architecture](https://github.com/lnis-uofu/OpenFPGA/blob/a9f91513f679c0285b6358a1fe04fb21d39b1ab5/openfpga_flow/vpr_arch/k6_frac_N10_tileable_adder_register_scan_chain_mem16K_depop50_12nm.xml#L135)

Reply:
Thanks. I have done changes accordingly